### PR TITLE
L1 cost fix

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: core/types/receipt.go
Comment (trianglesphere) [inline core/types/receipt.go:529]: This `time` is an internal non-consensus field used for spam detection. It is not going to be accurate here.

**Review by trianglesphere — CHANGES_REQUESTED** [review]

File: core/types/receipt.go
Comment (trianglesphere) [reply core/types/receipt.go:529]: I think you need to modify the `DeriveFields` function to take the timestamp. It's kinda annoying that we now use a time based fork in a place that doesn't keep references to time.

Comment (mslipper) [reply core/types/receipt.go:529]: Ah nice catch, didn't realize that. Fixed.

File: core/types/rollup_l1_cost.go
Comment (mslipper) [reply core/types/rollup_l1_cost.go:30]: Left as-is to minimize diff between this + Proto's PR.

Comment (mslipper) [reply core/types/rollup_l1_cost.go:30]: If you feel strongly about it I can fix.

**Review by trianglesphere — COMMENTED** [review]
changes lgtm. Will approve when tests pass.

**Review by trianglesphere — APPROVED** [review]

**Review by protolambda — APPROVED** [review]
LGTM. IMO worth squashing commits + adding more description. And before we merge we can do a `replace ../go-ethereum` locally into monorepo and see what tests break, and follow up with a monorepo PR to update any tests + the geth dependency. Not required for testnet testing though: the geth changes are isolated.

**Comment by mslipper** [comment]
Yep, will redo commit messages before merging. Thanks all!

**Pull request merged**
